aten/src/ATen/native/native_functions.yaml

```diff
@@ -13147,10 +13147,16 @@
 - func: linalg_lu_factor(Tensor A, *, bool pivot=True) -> (Tensor LU, Tensor pivots)
   python_module: linalg
   variants: function
+  dispatch:
+    CompositeImplicitAutograd: linalg_lu_factor
+    MPS: linalg_lu_factor_mps
 
 - func: linalg_lu_factor.out(Tensor A, *, bool pivot=True, Tensor(a!) LU, Tensor(b!) pivots) -> (Tensor(a!) LU, Tensor(b!) pivots)
   python_module: linalg
   variants: function
+  dispatch:
+    CompositeImplicitAutograd: linalg_lu_factor_out
+    MPS: linalg_lu_factor_out_mps
```

Comment:
Could you implement `linalg_lu_factor_ex` rather than this one? That way you wouldn't need to add any new backward rule, and all the other goodies that are implemented for `linalg.lu_factor`  will also extend to MPS.

Reply:
The reason I implement `linalg_lu_factor` is because `linalg_lu_factor_ex` has to return an `info` tensor, which is not applicable to MPS. (the info tensor is computed by LAPACK on cpu, for example).

Reply:
I guess I cannot directly return an undefined Tensor `Tensor()` as the `info` since there are some post check logics to the `info` tensor.